addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1404,6 +1411,21 @@ const Wysiwyg = Widget.extend({
     getInSelection(selector) {
         return getInSelection(this.odooEditor.document, selector);
     },
+    /**
+     * Adds an empty action in the mutex. Can be used to wait for some options
+     * to be initialized before doing something else.
+     *
+     * @returns {Promise}
+     */
+    waitForEmptyMutexAction() {
+        const self = this;
+        if (self.snippetsMenu) {
+            return new Promise(resolve => {
+                self.snippetsMenu.execWithLoadingEffect(() => null, false).then(() => resolve());
+            });
+        }
+        return Promise.resolve();
```

Comment:
- You don't need `self` if I am not mistaken, since you use an arrow function (as you should) anyway ;) Basically if you write new code (here a new function) that needs `self`, you are doing something wrong
- I don't think you need a new `Promise`? Can't you just return the `exec` call result?